lib/spack/spack/installer.py

```diff
@@ -1687,6 +1682,10 @@ def _install_task(self, task: BuildTask, install_status: InstallStatus) -> None:
         try:
             self._setup_install_dir(pkg)
 
+            # Create stage object now and let it be serialized for the child process. That
+            # way monkeypatch in tests works correctly.
+            pkg.stage
```

Comment:
This screams for a later refactor (attribute access with side effects :open_mouth:)

Reply:
True, but it's effectively `pkg.stage()`